src/io/image_olm_parser.rs

```diff
@@ -148,41 +163,47 @@ fn alias_sub_image(image: RgbImage, pixel_aliases: &PixelKeys) -> Vec<usize> {
 fn alias_pixels(image: &RgbImage) -> PixelKeys {
     image
         .pixels()
-        .fold(HashSet::<Rgb<u8>>::new(), |mut acc, pixel| {
-            acc.insert(*pixel);
+        .fold(Vec::<Rgb<u8>>::new(), |mut acc, pixel| {
+            acc.push(*pixel);
             acc
         })
         .into_iter()
+        .unique()
```

Comment:
It is correct that the algorithm should run deterministically with a given seed and I appreciate the desire to make this deterministic (perhaps inline with that principle), but I disagree with doing this for aliases. I disagree becasue functionally the association of a particular alias with a particular pixel should have **no** effect on the output, and so adding a layer of determinism here is a purely artifical addition without effect. Now this change by itself should not effect anything so maybe it does not matter by itself, but keeping this leaves functionality that may be relied upon in the future unintentionally. If we need this determinism then fine, but if it becomes relied upon unintentionality then, in my view, that's tech debt.